### ISSUE TITLE
S1128 False Positive when using System.Linq

### ISSUE THREAD

**Issue by jcurl** [SonarSource/sonar-dotnet#2758]
### Description

Visual Studio 2019 had SonarLint automatically upgrade to 4.13.0.11687. Warning S1128 (remove unused usings) false positive with `using System.Linq` and there's a Linq query in the code. It will raise a warning to remove the statement, resulting in a compilation error if removed.

### Repro steps

See the code

```csharp
using System.Linq;

[Test]
public void TestCase {
  List section = new List<string>() {
    "key";
  }
  var copiedKeys = from item in array select item.Key;
  Assert.That(copiedKeys, Has.Member("key"));
}

```

### Expected behavior

That SonarLint is able to know there is a Linq query, and that it does not warn about removing `System.Linq`.

### Actual behavior

The warning S1128 is shown, when it shouldn't be, and it is removed in a bulk change.

### Known workarounds

None

### Related information

* SonarC# Version: Using SonarLint 4.13.0.11687
* Visual Studio Version 2019 version 16.3.6
* If running through the Scanner for MSBuild, its version: N/A
